restapi/src/main/java/io/stargate/web/docsapi/service/write/DocumentWriteService.java

```diff
@@ -237,6 +237,71 @@ public DocumentWriteService(TimeSource timeSource, DocsApiConfiguration config)
         .observeOn(Schedulers.io());
   }
 
+  /**
+   * Deletes a single whole document.
+   *
+   * @param dataStore {@link DataStore}
+   * @param keyspace Keyspace to delete a document from.
+   * @param collection Collection the document belongs to.
+   * @param documentId Document ID.
+   * @param rows Rows of this document.
+   * @param numericBooleans If numeric boolean should be stored.
+   * @param context Execution content for profiling.
+   * @return Single containing the {@link ResultSet} of the batch execution.
+   */
+  public Single<ResultSet> deleteDocument(
+      DataStore dataStore,
+      String keyspace,
+      String collection,
+      String documentId,
+      ExecutionContext context) {
+    List<String> subDocumentPath = Collections.emptyList();
+    return deleteDocument(dataStore, keyspace, collection, documentId, subDocumentPath, context);
+  }
+
+  /**
+   * Deletes a single (sub-)document, ensuring that existing document with the same key have the
+   * given sub-path deleted.
+   *
+   * @param dataStore {@link DataStore}
+   * @param keyspace Keyspace to delete a document from.
+   * @param collection Collection the document belongs to.
+   * @param documentId Document ID.
+   * @param subDocumentPath The sub-document path to delete.
+   * @param context Execution content for profiling.
+   * @return Single containing the {@link ResultSet} of the batch execution.
+   */
+  public Single<ResultSet> deleteDocument(
+      DataStore dataStore,
+      String keyspace,
+      String collection,
+      String documentId,
+      List<String> subDocumentPath,
+      ExecutionContext context) {
+    // create and cache the remove query prepare
+    Single<? extends Query<? extends BoundQuery>> deleteQueryPrepare =
+        prepareDeleteDocumentQuery(dataStore, keyspace, collection, subDocumentPath);
+
+    // execute when prepared
+    return deleteQueryPrepare
+        .observeOn(Schedulers.computation())
+        .map(
+            prepared -> {
+              // take current timestamp for binding
+              long timestamp = timeSource.currentTimeMicros();
+
+              // return bind delete
+              // TODO Eric I feel keeping deletes always with the timestamp - 1 feels right
```

Comment:
Yes this is alright - I know that when writing chunks of documents there is a delete statement in the batch that has `timestamp - 1` as its value, so to keep it consistent this should be fine